libtransmission/global-ip-cache.cc

```diff
@@ -84,7 +84,7 @@ namespace global_source_ip_helpers
                 auto src_sslen = socklen_t{ sizeof(src_ss) };
                 if (getsockname(sock, reinterpret_cast<sockaddr*>(&src_ss), &src_sslen) == 0)
                 {
-                    if (auto const addrport = tr_socket_address::from_sockaddr(reinterpret_cast<sockaddr*>(&src_ss)); addrport)
+                    if (auto const addrport = tr_socket_address::from_sockaddr(reinterpret_cast<sockaddr*>(&src_ss)))
```

Comment:
This feels too close to the good old "assignment within a conditional expression" type of warning, not sure I like it.

Reply:
c1906837b050d76af4311e6b6d3d0358946c0981